src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -2368,7 +2368,16 @@ protected static bool TypesEquivalent(ITypeSymbol? oldType, ITypeSymbol? newType
             => oldTypes.SequenceEqual(newTypes, exact, (x, y, exact) => TypesEquivalent(x, y, exact));
 
         protected static bool ParameterTypesEquivalent(IParameterSymbol oldParameter, IParameterSymbol newParameter, bool exact)
-            => (exact ? s_exactSymbolEqualityComparer : s_runtimeSymbolEqualityComparer).ParameterEquivalenceComparer.Equals(oldParameter, newParameter);
+        {
+            if (exact)
+            {
+                return s_exactSymbolEqualityComparer.ParameterEquivalenceComparer.Equals(oldParameter, newParameter);
+            }
+
+            return s_runtimeSymbolEqualityComparer.ParameterEquivalenceComparer.Equals(oldParameter, newParameter) ||
```

Comment:
>Equals [](http://example.com/codeflow?start=80&length=6)

Equals has an overload that takes `bool compareParameterName, bool isCaseSensitive` so perhaps we could use it?

Reply:
They both default to false so they're  already doing what we want here. Can specify if you think it's clearer though?

Or did you mean put the discard checking into the parameter equality conparer?

Reply:
I meant pass `compareParameterName: exact`

Reply:
If the comparer is not checking discards then we should also fix that.

Reply:
> I meant pass compareParameterName: exact

I'll try it and see what breaks. This bit of code isn't actually needed for parameter renames, this is just for lambda discards, as a discards type info is different.

Reply:
Oh, interesting - so discards are not just synthesized names?

Reply:
Ignore me, it turns out this change isn't actually unnecessary, I just had a bad test that had a compile error. I was going from `(int a, int b)` to `(int a, _)`, but that's not actually legal.

Reply:
Change `int a` to `_`, oldSymbol vs newSymbol:
![image](https://user-images.githubusercontent.com/754264/127434712-369ac2d0-6a6b-44b5-8f54-c4ab8208ae28.png)